### DISCUSSION TITLE
What is the limit of connection pool size for s3 client?

### DISCUSSION THREAD

**Discussion by sfc-gh-schen** [aws/aws-sdk-java#3065] [Q&A] [answered]
Per this doc https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/ClientConfiguration.html#DEFAULT_MAX_CONNECTIONS
The default value of the max connection pool size is 50, what is the upper limit for it?

**Comment by debora-ito** [comment, marked as answer]
In theory, the upper limit of max connections is the maximum number of file descriptors your operating system can have. 

Note that increasing the number of concurrent connections can increase client throughput, but it also means more resources your application has to manage simultaneously.

**Reply by sfc-gh-schen** [reply]
Thank you Debora. This is helpful information :)
